central/complianceoperator/v2/report/manager/watcher/validator.go

```diff
@@ -0,0 +1,119 @@
+package watcher
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	complianceIntegrationDS "github.com/stackrox/rox/central/complianceoperator/v2/integration/datastore"
+	"github.com/stackrox/rox/central/complianceoperator/v2/report"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/set"
+)
+
+// ValidateScanConfigResults returns a map with the clusters that failed to be scanned.
+func ValidateScanConfigResults(ctx context.Context, results *ScanConfigWatcherResults, integrationDataStore complianceIntegrationDS.DataStore) (map[string]*storage.ComplianceOperatorReportSnapshotV2_FailedCluster, error) {
+	failedClusters := make(map[string]*storage.ComplianceOperatorReportSnapshotV2_FailedCluster)
+	errList := errorhelpers.NewErrorList("failed clusters")
+	failedClusterSet := set.NewStringSet()
+	successfulClusterSet := set.NewStringSet()
+	for _, scanResult := range results.ScanResults {
+		failedClusterSet.Add(scanResult.Scan.GetClusterId())
```

Comment:
`failedClusterSet` will contain all clusters - is that the intention, or should this be moved after:
```
		if failedClusterInfo == nil {
			successfulClusterSet.Add(scanResult.Scan.GetClusterId())
			continue
		}
```

**EDIT:** after looking at the logic, we are using `failedClusterSet` and `successfulClusterSet` only to see if we already had that cluster in the results. Maybe we can use one set? (`visitedClusterSet` or `clustersWithResults`)

Reply:
Good catch! `successfulClusterSet` was completely redundant and `failedClusterSet` had the incorrect name. I've changed to `clustersWithResults`.